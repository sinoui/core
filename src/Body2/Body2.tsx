import { TextContent, typographyPub } from '../utils/typographyPub';

const Body2 = typographyPub('p', 'body2')(TextContent);
export default Body2;
