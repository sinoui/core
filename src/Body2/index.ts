import Body2 from './Body2';

export default Body2;
