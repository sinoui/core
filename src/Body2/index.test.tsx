import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import Body2 from './index';

/**
 * Body2组件单元测试
 */
describe('文字排版 单元测试', () => {
  afterEach(cleanup);

  it('指定不同的根元素', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Body2 as="div">这是一个文字。</Body2>
      </ThemeProvider>,
    );

    const text = container.querySelector('div');
    expect(text).toHaveTextContent('这是一个文字。');
  });
  it('通过 paragraphy 指定下边距', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Body2 paragraph>这是一个文字。</Body2>
      </ThemeProvider>,
    );

    const text = container.querySelector('.sinoui-typography--paragraph');
    expect(text).toHaveTextContent('这是一个文字。');
  });

  it('ref 指向根元素', () => {
    const ref: React.RefObject<HTMLInputElement> = React.createRef();
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Body2 data-testid="typo" ref={ref}>
          这是一个文字。
        </Body2>
      </ThemeProvider>,
    );

    const text = container.firstChild;
    expect(text).toEqual(ref.current);
  });
  it('设置align文本对齐方式', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Body2 data-testid="typo" align="center">
          这是一个文字。
        </Body2>
      </ThemeProvider>,
    );

    const text = container.firstChild;
    expect(text).toHaveStyle(`text-align:center`);
  });
});

/**
 * Body2组件快照测试
 */
describe('文字排版 快照测试', () => {
  it('基本文字排版', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Body2>Body2</Body2>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('设置属性', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Body2 as="h2">Body2</Body2>
          <Body2 align="center">Body2</Body2>
          <Body2 noWrap>Body2</Body2>
          <Body2 color="success">Body2</Body2>
          <Body2 paragraphy>Body2</Body2>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('颜色设置', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Body2>Body2</Body2>
          <Body2 color="primary">Body2</Body2>
          <Body2 color="secondary">Body2</Body2>
          <Body2 color="textPrimary">Body2</Body2>
          <Body2 color="textSecondary">Body2</Body2>
          <Body2 color="error">Body2</Body2>
          <Body2 color="warning">Body2</Body2>
          <Body2 color="success">Body2</Body2>
          <Body2 color="info">Body2</Body2>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
