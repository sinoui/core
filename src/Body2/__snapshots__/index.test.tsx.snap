// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`文字排版 快照测试 基本文字排版 1`] = `
<p
  className="sc-AxjAm hmEnYx sinoui-typography sinoui-typography--body2"
  color="textPrimary"
  type="body2"
>
  Body2
</p>
`;

exports[`文字排版 快照测试 设置属性 1`] = `
Array [
  <h2
    className="sc-AxjAm hmEnYx sinoui-typography sinoui-typography--body2"
    color="textPrimary"
    type="body2"
  >
    Body2
  </h2>,
  <p
    className="sc-AxjAm kGGcFZ sinoui-typography sinoui-typography--body2"
    color="textPrimary"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm iSfNrw sinoui-typography sinoui-typography--body2"
    color="textPrimary"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm iIaOmZ sinoui-typography sinoui-typography--body2"
    color="success"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm hmEnYx sinoui-typography sinoui-typography--body2"
    color="textPrimary"
    type="body2"
  >
    Body2
  </p>,
]
`;

exports[`文字排版 快照测试 颜色设置 1`] = `
Array [
  <p
    className="sc-AxjAm hmEnYx sinoui-typography sinoui-typography--body2"
    color="textPrimary"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm iFWQyW sinoui-typography sinoui-typography--body2"
    color="primary"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm eSSXbd sinoui-typography sinoui-typography--body2"
    color="secondary"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm hmEnYx sinoui-typography sinoui-typography--body2"
    color="textPrimary"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm fgkYSM sinoui-typography sinoui-typography--body2"
    color="textSecondary"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm fAjjbc sinoui-typography sinoui-typography--body2"
    color="error"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm cWvbUV sinoui-typography sinoui-typography--body2"
    color="warning"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm iIaOmZ sinoui-typography sinoui-typography--body2"
    color="success"
    type="body2"
  >
    Body2
  </p>,
  <p
    className="sc-AxjAm cPGedZ sinoui-typography sinoui-typography--body2"
    color="info"
    type="body2"
  >
    Body2
  </p>,
]
`;
